packages/databyss-services/topics/actions.js

```diff
@@ -0,0 +1,45 @@
+import * as topics from './'
+
+import { FETCH_TOPIC, SAVE_TOPIC, CACHE_TOPIC } from './constants'
+
+export function fetchTopic(id) {
+  return async dispatch => {
+    dispatch({
+      type: FETCH_TOPIC,
+      payload: {},
+    })
+
+    topics
+      .getTopic(id)
+      .then(topic => {
+        dispatch({
+          type: CACHE_TOPIC,
+          payload: { topic, id },
+        })
+      })
+      .catch(err => {
+        dispatch({
+          type: CACHE_TOPIC,
+          payload: {
+            topic: err,
+            id,
+          },
+        })
+      })
+  }
+}
+
+export function saveTopic(topicFields) {
+  return async dispatch => {
+    dispatch({
+      type: SAVE_TOPIC,
+      payload: { topic: topicFields, id: topicFields._id },
+    })
+    topics.setTopic(topicFields).then(() => {
+      dispatch({
```

Comment:
We don't need a `CACHE_TOPIC` here because `SAVE_TOPIC` already updates the cache.